src/blockchain/Blockchain.ts

```diff
@@ -141,28 +211,158 @@ export default class Blockchain implements IBlockchainApi {
+   * finalization and rejects any errors occur during submission or execution of extrinsics. This behavior can be adjusted via optional parameters.
+   *
+   * This function will try to re-sign and resend the given transaction two times recursively, if recoverable errors are encountered.
+   *
+   * Transaction fees will apply whenever a transaction fee makes it into a block, even if extrinsics fail to execute correctly!
+   *
+   * @param identity The [[SubmittableExtrinsic]] to be submitted. Most transactions need to be signed, this must be done beforehand.
+   * @param tx The [[SubmittableExtrinsic]] to be submitted. Most transactions need to be signed, this must be done beforehand.
+   * @param opts Criteria for resolving/rejecting the promise.
+   * @returns A promise which can be used to track transaction status.
+   * If resolved, this promise returns [[SubmittableResult]] that has led to its resolution.
+   */
+  public async submitSignedTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic,
+    opts?: SubscriptionPromiseOptions
+  ): Promise<SubmittableResult> {
+    const options = parseSubscriptionOptions(opts)
+    return submitSignedTx(tx, options)
+      .catch(async (reason: Error) => {
+        if (reason.message === 'Recoverable') {
+          return submitSignedTx(await this.reSignTx(identity, tx), options)
+        }
+        throw reason
+      })
+      .catch(async (reason: Error) => {
+        if (reason.message === 'Recoverable') {
+          return submitSignedTx(await this.reSignTx(identity, tx), options)
+        }
+        throw reason
+      })
+  }
+
+  /**
+   * [ASYNC] Signs and submits the SubmittableExtrinsic with optional resolving and rejection criteria.
+   *
+   * @param identity The [[Identity]] that we retrieve the nonce for.
+   * @param tx The generated unsigned SubmittableExtrinsic to submit.
+   * @param opts Optional [[SubscriptionPromiseOptions]].
+   * @returns Promise Result of The Extrinsic.
+   *
+   */
   public async submitTx(
     identity: Identity,
     tx: SubmittableExtrinsic,
-    opts: SubscriptionPromiseOptions = {}
+    opts?: SubscriptionPromiseOptions
   ): Promise<SubmittableResult> {
     const signedTx = await this.signTx(identity, tx)
-    return Blockchain.submitSignedTx(signedTx, opts)
+    return this.submitSignedTx(identity, signedTx, opts)
+  }
+
+  /**
+   * [ASYNC] Retrieves the Nonce for Transaction signing for the specified account, and increments the in accountNonces mapped Index.
+   *
+   * @param accountAddress The address of the identity that we retrieve the nonce for.
+   * @param reset Specify whether the entry for the account is outdated and has to be reset.
+   * @returns [[Index]] representation of the Tx nonce for the identity.
+   *
+   */
+  public async getNonce(accountAddress: string, reset = false): Promise<Index> {
+    if (reset) {
+      this.accountNonces.delete(accountAddress)
+    }
```

Comment:
maybe we could move that to a separate method? The call side could just call `resetNonce(account)`. That way the method only does one thing , which reduces complexity.

Reply:
Agreed.

Reply:
As we currently only reset inside reSignTx, we reset it there.